posting/index.go

```diff
@@ -1387,6 +1386,37 @@ func rebuildTokIndex(ctx context.Context, rb *IndexRebuild) error {
 	builder.fn = func(uid uint64, pl *List, txn *Txn) ([]*pb.DirectedEdge, error) {
 		edge := pb.DirectedEdge{Attr: rb.Attr, Entity: uid}
 		edges := []*pb.DirectedEdge{}
+
+		if runForVectors {
+			val, err := pl.Value(txn.StartTs)
+			if err != nil {
+				return []*pb.DirectedEdge{}, err
+			}
+
+			if val.Tid != types.VFloatID {
+				sv, err := types.Convert(val, types.VFloatID)
+				if err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+				b := types.ValueForType(types.BinaryID)
+				if err = types.Marshal(sv, &b); err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+				edge.Value = b.Value.([]byte)
+				edge.ValueType = types.VFloatID.Enum()
+
+				inKey := x.DataKey(edge.Attr, uid)
+				p, err := txn.Get(inKey)
+				if err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+
+				if err := p.addMutation(ctx, txn, &edge); err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+			}
+		}
+
 		err := pl.Iterate(txn.StartTs, 0, func(p *pb.Posting) error {
```

Comment:
No need to iterate when you already have the value in val. Just run the function inside iterate